Index: provider.c
===================================================================
--- provider.c	(revision 308495)
+++ provider.c	(revision 308598)
@@ -424,6 +424,8 @@
 		if(proto && Z_STRLEN_PP(proto)) {	
 			spprintf(&tmp, 0, "%s://%s%s", Z_STRVAL_PP(proto), Z_STRVAL_PP(host), Z_STRVAL_PP(uri));
 		} else {
+			SEPARATE_ZVAL(port);
+			convert_to_long_ex(port);
 			spprintf(&tmp, 0, "http%s://%s%s", Z_LVAL_PP(port)==443?"s":"", Z_STRVAL_PP(host), Z_STRVAL_PP(uri));
 		}
 		return tmp;
